spec/menubar/utils_spec.lua

```diff
@@ -65,6 +65,65 @@ describe("menubar.utils lookup_icon_uncached", function()
         assert_found_in_path('awesome',  '/.icons/awesome/64x64/apps/awesome.png')
         assert_found_in_path('awesome2', '/.icons/awesome/scalable/apps/awesome2.png')
     end)
+
+-- Check for no argument :: if not icon_file or icon_file == "" then
+    it('no icon specified', function()
+        assert.is_false(utils.lookup_icon_uncached())
+        assert.is_false(utils.lookup_icon_uncached(nil)) -- Same as above?
+        assert.is_false(utils.lookup_icon_uncached(false))
+        assert.is_false(utils.lookup_icon_uncached(''))
+    end)
+
+-- Check for icon not in search path :: if icon_file:sub(1, 1) == '/' and supported_icon_formats[icon_file_ext] then
+    it('finds icons even those not in the search paths when full path specified', function()
+
+        -- Shimmed icon base directories contain the following icons:
+        --
+        --     usr/share/icon5.png
+        --     usr/share/icon6.xpm
+        --     usr/share/icon7.svg
+
+        assert_found_in_path('/usr/share/icon5.png', '/usr/share/icon5.png')
+        assert_found_in_path('/usr/share/icon6.xpm', '/usr/share/icon6.xpm')
+        assert_found_in_path('/usr/share/icon7.svg', '/usr/share/icon7.svg')
```

Comment:
ok, you moved root to the top, it was half of what need to be done

now use it to fix your assertions in this test

Reply:
AFAIA the asserts are what they need to be...

ie: assert.matches compares
a fixed string (the left) with the value returned from `utils.lookup_icon_uncached(icon)` which should allow '/usr/share/icon5.png'

ie: why does this one `assert_found_in_path('icon5.png', '/.icons/icon5.png')` work but not the next two?

Anyway - I won't argue the point, I'll just do what you suggest

Reply:
Still no good -  `utils.lookup_icon_uncached(icon)` still returning ''

Failure -> spec/menubar/utils_spec.lua @ 78
menubar.utils lookup_icon_uncached finds icons even those not in the search paths when full path specified
spec/menubar/utils_spec.lua:17: Expected strings to match.
Passed in:
(string) ''
Expected:
(string) '/spec/menubar/usr/share/icon5.png$'
stack traceback:
	spec/menubar/utils_spec.lua:17: in function 'assert_found_in_path'
	spec/menubar/utils_spec.lua:86: in function <spec/menubar/utils_spec.lua:78>

Reply:
Wrong